flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/sink/serializer/AvroToProtoSerializerTest.java

```diff
@@ -16,20 +16,731 @@
 
 package com.google.cloud.flink.bigquery.sink.serializer;
 
+import com.google.cloud.flink.bigquery.sink.exceptions.BigQuerySerializationException;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Descriptors.Descriptor;
+import com.google.protobuf.Descriptors.FieldDescriptor;
+import com.google.protobuf.DynamicMessage;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericFixed;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.generic.GenericRecordBuilder;
+import org.assertj.core.api.Assertions;
 import org.junit.Test;
 
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static com.google.cloud.flink.bigquery.sink.serializer.AvroToProtoSerializer.getDynamicMessageFromGenericRecord;
+import static com.google.cloud.flink.bigquery.sink.serializer.TestBigQuerySchemas.getAvroSchemaFromFieldString;
+import static com.google.cloud.flink.bigquery.sink.serializer.TestBigQuerySchemas.getRecordSchema;
 import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertThrows;
 
 /** Tests for {@link AvroToProtoSerializer}. */
 public class AvroToProtoSerializerTest {
 
     @Test
-    public void testSerializeIsUnsupported() {
-        UnsupportedOperationException exception =
+    public void testPrimitiveTypesConversion() {
+        BigQuerySchemaProvider bigQuerySchemaProvider =
+                TestBigQuerySchemas.getSchemaWithRequiredPrimitiveTypes();
+        BigQueryProtoSerializer<GenericRecord> serializer = new AvroToProtoSerializer();
+        serializer.init(bigQuerySchemaProvider);
+        Descriptor descriptor = bigQuerySchemaProvider.getDescriptor();
+        Schema avroSchema = bigQuerySchemaProvider.getAvroSchema();
+
+        byte[] byteArray = "Any String you want".getBytes();
+        Schema recordSchema = avroSchema.getField("required_record_field").schema();
+        GenericRecord record =
+                new GenericRecordBuilder(avroSchema)
+                        .set("number", -7099548873856657385L)
+                        .set("price", 0.5616495161359795)
+                        .set("species", "String")
+                        .set("flighted", true)
+                        .set("sound", ByteBuffer.wrap(byteArray))
```

Comment:
should this cover all the primitive types that we support? for example, what about enum?

Reply:
Yep, we do test for all primitive types offered by avro. [Primitive Types](https://github.com/prashastia/flink-bigquery-connector/blob/7af9b7190ae53afb25e5f9c1aca2a610a5f40388/flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/sink/serializer/TestBigQuerySchemas.java#L77) tests for the Datatypes present in BQ, while [Remaining Primitive Type ](https://github.com/prashastia/flink-bigquery-connector/blob/7af9b7190ae53afb25e5f9c1aca2a610a5f40388/flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/sink/serializer/TestBigQuerySchemas.java#L103) checks for other primitive types like ENUM (not present in BQ)

Reply:
ah okay sorry missed those test; thanks! you can resolve this convo